pkg/build/api/v1/types.go

```diff
@@ -402,6 +408,10 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// ExternalBuildStrategy defines input parameters specific to an External build.
+type ExternalBuildStrategy struct {
```

Comment:
The URL is likely to only be known after the actual build job is created in the external system. We definitely want to be able to link stuff together but wondering if that is done better via annotations or as fields in this struct? Struct field probably better to make it properly specified & versionable.